lib/modules/datasource/maven/util.ts

```diff
@@ -130,6 +180,64 @@ export async function checkHttpResource(
   }
 }
 
+async function checkS3Resource(
+  pkgUrl: URL | string
+): Promise<HttpResourceCheckResult> {
+  try {
+    const s3Url = parseS3Url(pkgUrl.toString());
+    // istanbul ignore next
+    if (s3Url === null) {
+      return 'error';
+    }
+    const response = await getS3Client().headObject(s3Url);
+    // istanbul ignore next
+    if (response.DeleteMarker) {
+      return 'not-found';
+    }
+    if (response.LastModified) {
+      return response.LastModified;
+    }
+    return 'found';
+  } catch (err) {
+    if (isS3NotFound(err)) {
+      return 'not-found';
+    } /* istanbul ignore else */ else {
+      // istanbul ignore next
+      logger.debug(
+        { pkgUrl, name: err.name, message: err.message },
+        `Can't check S3 resource existence`
+      );
+    }
+    // istanbul ignore next
+    return 'error';
+  }
+}
+
+export async function checkResource(
+  http: Http,
+  pkgUrl: URL | string
+): Promise<HttpResourceCheckResult> {
+  const parsedUrl = typeof pkgUrl === 'string' ? parseUrl(pkgUrl) : pkgUrl;
+  // istanbul ignore next
```

Comment:
create test case, function ius exported, so can directly tested

Reply:
I added a cause explaining it's a null guard.